worker/modelcache/worker.go

```diff
@@ -181,6 +191,22 @@ func (c *cacheWorker) loop() error {
 					}
 				}
 			}
+
+			// In theory we could just create a new controller and just apply
+			// the new delta's to that. We can track if a new controller is
+			// required in the same vein as sweepRequired.
+			c.mu.Lock()
+			if c.sweepRequired {
```

Comment:
Do we know that the initial model delta dump comes through as the initial set? I think it probably does, but worth double checking.

Reply:
According to the [documentation](https://github.com/juju/juju/blob/develop/state/multiwatcher.go#L62-L70) and [looking](https://github.com/juju/juju/blob/develop/state/multiwatcher.go#L229) [through](https://github.com/juju/juju/blob/develop/state/allwatcher.go#L1443) the code, along local testing, this is the case.

Reply:
That's nice then :smile: